io-engine/src/host/blk_device.rs

```diff
@@ -278,7 +277,10 @@ fn new_device(
 fn get_mounts() -> Result<HashMap<OsString, Vec<MountInfo>>, Error> {
     let mut table: HashMap<OsString, Vec<MountInfo>> = HashMap::new();
 
-    for mount in (MountIter::new()?).flatten() {
+    for mount in SafeMountIter::get()
+        .map_err(|err| Error::new(ErrorKind::Other, err))?
```

Comment:
Isn't there a conversion already?

```suggestion
        ?
```

Reply:
No there isn't.. The `MountIter::new()` returned an io::Error, but this `SafeMountIter::get()` returns a MountInfoError. Ref: https://github.com/openebs/mayastor-dependencies/blob/develop/devinfo/src/mountinfo/error.rs#
```sh
error[E0277]: `?` couldn't convert the error to `std::io::Error`

   --> io-engine/src/host/blk_device.rs:280:38

    |

277 | fn get_mounts() -> Result<HashMap<OsString, Vec<MountInfo>>, Error> {

    |                    ------------------------------------------------ expected `std::io::Error` because of this

...

280 |     for mount in SafeMountIter::get()?.flatten() {

    |                                      ^ the trait `std::convert::From<devinfo::mountinfo::error::MountInfoError>` is not implemented for `std::io::Error`

    |

    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait

    = help: the following other types implement trait `std::convert::From<T>`:

              <std::io::Error as std::convert::From<rand::Error>>

              <std::io::Error as std::convert::From<getrandom::error::Error>>

              <std::io::Error as std::convert::From<httpdate::Error>>

              <std::io::Error as std::convert::From<prost::error::DecodeError>>

              <std::io::Error as std::convert::From<serde_json::Error>>

              <std::io::Error as std::convert::From<prost::error::EncodeError>>

              <std::io::Error as std::convert::From<nix::errno::Errno>>

              <std::io::Error as std::convert::From<tokio_stream::Elapsed>>

            and 8 others

    = note: required for `std::result::Result<std::collections::HashMap<std::ffi::OsString, std::vec::Vec<devinfo::mountinfo::MountInfo>>, std::io::Error>` to implement `std::ops::FromResidual<std::result::Result<std::convert::Infallible, devinfo::mountinfo::error::MountInfoError>>`
```

Reply:
Maybe we need a conversion from MountInfoError to Error then?

Reply:
Yep, good one. Done.